### PR TITLE
fix(react): update input value for SearchField in controlled case

### LINKED ISSUE #1432: Bug: Controlled SearchField
### Steps to reproduce

Steps to reproduce
1. Create a test component:
``` ts
function ExampleComponent() {
  const [searchField, setSearchField] = useState("");
  const [textField, setTextField] = useState("");

  return (
    <>
      <SearchField
        label="Find Rule"
        placeholder="Find Rule..."
        hideLabel
        value={searchField}
        onChange={(value, e) => {
          console.log("search field", value, e.target.value);
          setSearchField(value);
        }}
      />

      <TextField
        label="Find Rule"
        placeholder="Find Rule..."
        value={textField}
        onChange={(value, e) => {
          console.log("text field", value, e.target.value);
          setTextField(value);
        }}
      />
    </>
  );
}
```
2. Run example
3. Enter text `img` to both `SearchField` and `TextField`

### Expected Behavior

1. `SearchField` and `TextField` both contain text "img" that we've just entered
2. `searchField` and `textField` state contain string "img"

### Actual Behavior

1. `SearchField` does not contain any text entered, `TextField` works as expected
image
2. `searchField` state contains string "g", `textField` state contains string "img"

### Version

6.3.1

### What browsers are you experiencing the problem on?

_No response_

### Relevant log output

```shell
search field i i
search field m m
search field g g
text field i i
text field im im
text field img img
```

### Anything else we should know?

Additional information

### PR DESCRIPTION
**Pull request by chornonoh-vova**
closes #1432

### CODE REVIEW

File: docs/pages/components/SearchField.mdx [hunk 1]
```diff
@@ -72,6 +73,24 @@ We include trailing children to allow for additional input context to be provide
 </FieldWrap>
 ```
 
+### Controllable
```
Comment (scurker) [inline docs/pages/components/SearchField.mdx:76]: Nit: We've been calling this "controlled" in documentation.

```suggestion
### Controlled

`SearchField` can also be controlled, by setting the `value` prop and updating the value from event callbacks.
```

File: docs/pages/components/SearchField.mdx [hunk 2]
```diff
@@ -72,6 +73,24 @@ We include trailing children to allow for additional input context to be provide
...
+
+```jsx example
+function ControllableSearchFieldExample() {
```
Comment (scurker) [inline docs/pages/components/SearchField.mdx:79]:
```suggestion
function ControlledSearchFieldExample() {
```

File: docs/pages/components/SearchField.mdx [hunk 3]
```diff
@@ -72,6 +73,24 @@ We include trailing children to allow for additional input context to be provide
...
+  return (
+    <FieldWrap>
+      <SearchField
+        label="Controllable"
+        value={search}
+        onChange={(value) => setSearch(value)}
+      />
+    </FieldWrap>
+  );
+}
```
Comment (scurker) [inline docs/pages/components/SearchField.mdx:91]: Just some minor nits to bring it in line with other component documentation:

```suggestion
  const [value, setValue] = useState('Banana');
  const handleChange = (value) => {
    setValue(value);
  };
  
  return (
    <FieldWrap>
      <SearchField
        label="Controlled"
        value={value}
        onChange={handleChange}
      />
    </FieldWrap>
  );
}
```

**Review by scurker — CHANGES_REQUESTED** [review]
Mostly documentation nits and a few minor changes needed, but code LGTM.

File: docs/pages/components/SearchField.mdx (hunk 1 above)
Comment (chornonoh-vova) [reply docs/pages/components/SearchField.mdx]: Thanks, updated

File: docs/pages/components/SearchField.mdx (hunk 2 above)
Comment (chornonoh-vova) [reply docs/pages/components/SearchField.mdx]: Updated

File: docs/pages/components/SearchField.mdx (hunk 3 above)
Comment (chornonoh-vova) [reply docs/pages/components/SearchField.mdx]: Updated

**Review by scurker — APPROVED** [review]

**Pull request merged**